common/sgx_workload/workload/workload_processor.h

```diff
@@ -106,3 +130,23 @@ class WorkloadProcessor {
 #define REGISTER_WORKLOAD_PROCESSOR(WORKLOADID_STR,TYPE) \
    WorkloadProcessor* TYPE##_myProcessor = \
       WorkloadProcessor::RegisterWorkloadProcessor(WORKLOADID_STR, new TYPE());
+
+/**
+ * This macro registers a workload processor for a specific application
+ * and instantiates extended workorder interface appropriate to workload type.
+ * It associates a string with a workload.
+ * This is the same string that is passed in the work order request
+ * JSON payload.
+ * Example usage in a .cpp source file:
+ * REGISTER_WORKLOAD_PROCESSOR(workload_id_string, Workload)
+ *
+ * @param WORKLOADID_STR A string literal or variable identifying the workload
+ *                       type
+ * @param WORKLOAD_TYPE  Type of workload. Supported values are SINGLETON,
+ *                  KEY_MANAGEMENT_ENCLAVE and WORKORDER_MANAGEMENT_ENCLAVE
+ * @param TYPE           Name of the Workload class
+ */
+#define REGISTER_WORKLOAD_PROCESSOR_EXT_WORK_ORDER(WORKLOADID_STR,WORKLOAD_TYPE,TYPE) \
```

Comment:
Please use variadic macro to select between different version of `REGISTER_WORKLOAD_PROCESSOR` macro. This will be more user friendly for clinets writing workloads
```
#define GET_WORKLOAD_PROCESSOR(_1,_2,_3,NAME,...) NAME

#define REGISTER_WORKLOAD_PROCESSOR(...) \
        GET_WORKLOAD_PROCESSOR(__VA_ARGS__, REGISTER_WORKLOAD_PROCESSOR3, REGISTER_WORKLOAD_PROCESSOR2)(__VA_ARGS__)

#define REGISTER_WORKLOAD_PROCESSOR2(WORKLOADID_STR,TYPE) \
        WorkloadProcessor* TYPE##_myProcessor = \
          WorkloadProcessor::RegisterWorkloadProcessor(WORKLOADID_STR, new TYPE());

#define REGISTER_WORKLOAD_PROCESSOR3(WORKLOADID_STR,WORKLOADTYPE,TYPE) \
        WorkloadProcessor* TYPE##_myProcessor = \
          WorkloadProcessor::RegisterWorkloadProcessor(WORKLOADID_STR, WORKLOADTYPE, new TYPE());
```

Reply:
Done. Thanks for sharing the snippet.